render-wasm/src/shapes/text.rs

```diff
@@ -416,21 +441,72 @@ impl From<[u8; RAW_PARAGRAPH_DATA_SIZE]> for RawParagraphData {
     }
 }
 
+impl RawTextLeafData {
+    pub fn from(bytes: &[u8]) -> Self {
+        let total_fills = u32::from_be_bytes([bytes[52], bytes[53], bytes[54], bytes[55]]) as usize;
+        let fills_size = total_fills * RAW_LEAF_FILLS_SIZE;
+        let start = RAW_LEAF_DATA_SIZE;
+        let end = RAW_LEAF_DATA_SIZE + fills_size;
+
+        let fills: Vec<Fill> = if total_fills > 0 {
+            parse_fills_from_bytes(&bytes[start..end], total_fills)
+        } else {
+            Vec::new()
+        };
+
+        Self {
+            font_style: bytes[0],
+            font_size: f32::from_be_bytes([bytes[4], bytes[5], bytes[6], bytes[7]]),
+            font_weight: i32::from_be_bytes([bytes[8], bytes[9], bytes[10], bytes[11]]),
+            font_id: [
+                u32::from_be_bytes([bytes[12], bytes[13], bytes[14], bytes[15]]),
+                u32::from_be_bytes([bytes[16], bytes[17], bytes[18], bytes[19]]),
+                u32::from_be_bytes([bytes[20], bytes[21], bytes[22], bytes[23]]),
+                u32::from_be_bytes([bytes[24], bytes[25], bytes[26], bytes[27]]),
+            ],
+            font_family: [bytes[28], bytes[29], bytes[30], bytes[31]],
+            font_variant_id: [
+                u32::from_be_bytes([bytes[32], bytes[33], bytes[34], bytes[35]]),
+                u32::from_be_bytes([bytes[36], bytes[37], bytes[38], bytes[39]]),
+                u32::from_be_bytes([bytes[40], bytes[41], bytes[42], bytes[43]]),
+                u32::from_be_bytes([bytes[44], bytes[45], bytes[46], bytes[47]]),
+            ],
+            text_length: u32::from_be_bytes([bytes[48], bytes[49], bytes[50], bytes[51]]),
+            total_fills: total_fills as u32,
+            fills,
+        }
+    }
+}
+
```

Comment:
We could probably leverage `transmute` for this, like we did with the RawFillData